src/main/java/uk/gov/hmcts/divorce/caseworker/service/print/AppliedForCoPrinter.java

```diff
@@ -0,0 +1,113 @@
+import static uk.gov.hmcts.divorce.notification.FormatUtil.formatId;
+
+@Component
+@Slf4j
+public class AppliedForCoPrinter {
+
+    public static final String NAME = "name";
+    public static final String DATE_D84_RECEIVED = "dateD84Received";
+    public static final String GRANTED_DATE = "grantedDate";
+    private static final String LETTER_TYPE_APPLIED_FOR_CO = "applied-for-co-letter";
+
+    @Autowired
+    private BulkPrintService bulkPrintService;
+
+    @Autowired
+    private CaseDataDocumentService caseDataDocumentService;
+
+    @Autowired
+    private Clock clock;
+
+    public void print(final CaseData caseData,
+                      final Long caseId,
+                      final Applicant applicant) {
+
+        generateAppliedForCoLetter(caseData, caseId, applicant);
+
+        final List<Letter> appliedForCoLetters = lettersWithDocumentType(
+            caseData.getDocuments().getDocumentsGenerated(),
+            APPLIED_FOR_CO_LETTER);
+
+        final Letter appliedForCoLetter = firstElement(appliedForCoLetters);
+
+        if (!isEmpty(appliedForCoLetter)) {
+
+            final String caseIdString = caseId.toString();
+            final Print print =
+                new Print(singletonList(appliedForCoLetter), caseIdString, caseIdString, LETTER_TYPE_APPLIED_FOR_CO);
+            final UUID letterId = bulkPrintService.print(print);
+
+            log.info("Letter service responded with letter Id {} for case {}", letterId, caseId);
+        } else {
+            log.warn(
+                "Applied for Conditional Order has missing documents. Expected document with type {} , for Case ID: {}",
+                List.of(APPLIED_FOR_CO_LETTER),
+                caseId
+            );
+        }
+    }
+
+    private void generateAppliedForCoLetter(final CaseData caseData,
+                                            final Long caseId,
+                                            final Applicant applicant) {
+
+        log.info("Generating coversheet for sole case id {} ", caseId);
+        caseDataDocumentService.renderDocumentAndUpdateCaseData(
+            caseData,
+            APPLIED_FOR_CO_LETTER,
+            templateContent(caseData, caseId, applicant),
+            caseId,
+            APPLIED_FOR_CONDITIONAL_ORDER_LETTER_TEMPLATE_ID,
+            applicant.getLanguagePreference(),
+            formatDocumentName(caseId, APPLIED_FOR_CONDITIONAL_ORDER_LETTER_DOCUMENT_NAME, now(clock))
+        );
+    }
+
+    private Map<String, Object> templateContent(final CaseData caseData,
+                                                final Long caseId,
+                                                final Applicant applicant) {
+
+        final Map<String, Object> templateContent = new HashMap<>();
+
+        templateContent.put(CASE_REFERENCE, formatId(caseId));
+        templateContent.put(NAME, join(" ", applicant.getFirstName(), applicant.getLastName()));
```

Comment:
picky but Applicant has a getFullName method which could be used?

Reply:
I don't want the middle names on this though